pkg/executor/statefulset.go

```diff
@@ -154,11 +155,31 @@ func getStatefulset(
 	persistDirs := make([]string, len(defaultMountFolders), len(defaultMountFolders)+len(appPersistDirs))
 	copy(persistDirs, defaultMountFolders)
 
+	// if the app directory is either same or a subdir of one of the default mount
+	// dirs, we can skip them. if not we should add them to the persistDirs list
 	for _, appDir := range appPersistDirs {
-		// If this app directory is not present in the default mounts
-		// append to the persistDirs list
-		if !shared.StringInList(appDir, defaultMountFolders) {
-			persistDirs = append(persistDirs, appDir)
+		isSubDir := false
+		for _, defaultDir := range defaultMountFolders {
```

Comment:
What if a dir is not a subdirectory of one of the defaults, but IS the subdirectory of one of the other specified dirs?

Reply:
We should add validation or even mutate to catch that i think no?

Reply:
Or we could ignore such things similarly as we ignore subdirs of defaults. I.e. treat it like: first compose a list that is a union of specified and defaults, and then discard anything from that list which is a subdir of something else in that list.

Reply:
👍